packages/core/src/components/grading-instructions/EditableGradingInstruction.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useContext, useEffect, useRef } from 'react'
+import { makeRichText } from 'rich-text-editor'
+import { GradingInstructionContext } from '../context/GradingInstructionContext'
+import { CommonExamContext } from '../context/CommonExamContext'
+import { QuestionContext } from '../context/QuestionContext'
+
+export function EditableGradingInstruction({ element }: { element: Element }) {
+  const { language } = useContext(CommonExamContext)
+  const { displayNumber } = useContext(QuestionContext)
+  const { onChange, saveScreenshot } = useContext(GradingInstructionContext)
+  const answerGradingInstructionDiv = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    if (answerGradingInstructionDiv.current) {
+      makeRichText(
+        answerGradingInstructionDiv.current,
+        {
+          locale: language.startsWith('sv') ? 'SV' : 'FI',
+          screenshotSaver: saveScreenshot ? ({ type, data }) => saveScreenshot(type, data, displayNumber) : undefined,
+          forceInit: true,
+          screenshotImageSelector: 'img[src^="data:image/png"], img[src^="data:image/jpeg"]',
+          fileTypes: ['image/png', 'image/jpeg']
+        },
+        ({ answerHTML }) => (onChange ? onChange(answerHTML, displayNumber) : () => {})
```

Comment:
on change ja save screenshot lienee pakollisia tässä komponentissa?

Reply:
Kyllä, mutta se hyödyntää GradingInstructionPropsia ja sinne en laittaisi noita pakollisiksi, kun hvp:t ei ole joka paikassa editoitavia eikä noilla callbackeilla ole silloin käyttöä.

Reply:
niin tää johtuu siitä että meidän GradingInstruction:lla on samat propsit kuin contextilla. Usein näin ei ole mutta meidän tapauksessa on. Tää on tää vanha juttu et onko oma tyyppi vai `?` parempi. Käy näinkin

Reply:
Ehkä ne voisi silti erottaa toisistaan. Katsotaan.

Reply:
muistaakseni typescriptiä voi kikkailla silleen, että onChange olisi pakollinen vain sillon kun editable on true.

Reply:
tarkoitan tätä  https://www.zhenghao.io/posts/ts-never#partially-disallow-structural-typing

jotenkin näin

```
type Editable= {
    editable: true
    onChange:(answerHTML: string, displayNumber?: string) => void
}

type NotEditable = {
    editable: false
    onChange: never
}
```

Reply:
En jotenkin lämpene tälle. Providerissa pitää eksplisiittisesti castata toinen noista tyypeistä.

Reply:
en ehkä ihan saa kiinni mitä tarkotat mutta eikö siinä voisi tehdä jonkun typeguardin missä määritetään mit contextille annetaan esim 

```
  const contextValue = editable ? { editable, onChange, saveScreenshot } : { editable, saveScreenshot }
  return <GradingInstructionContext.Provider value={contextValue}>{children}</GradingInstructionContext.Provider>
 ```

Reply:
Pikaisella kokeilulla sain:

Types of property editable are incompatible.
Type boolean is not assignable to type false

En selvitellyt enempää.

Reply:
Selvitin kuitenkin, sillä on eroa, käyttääkö `type`ä vai `interface`a.

Reply:
mä pullasin tän haaran ja sain sen jotenkin tyypitettyä, voidaan katsoa myös yhdessä.